test/functional/fixtures/regression/gh-423/pages/index.html

```diff
@@ -0,0 +1,157 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>GH-423</title>
+</head>
+<body>
+<div id="target1" style="background: green;position: absolute; width:100px; height:100px;">
+</div>
+<div id="element" style="top:200px;background: red;position: absolute; width:100px; height:100px;">
+</div>
+<div id="target2" style="left: 150px;background: silver;position: absolute; width:100px; height:100px;">
+</div>
+<div id="child1" style="left: 150px;top:200px;background: blue;position: absolute; width:100px; height:100px;">
+</div>
+<div id="target3" style="left: 300px;background: violet;position: absolute; width:50px; height:50px;">
+</div>
+<div id="child2" style="left: 300px;top:200px;background: yellow;position: absolute; width:100px; height:100px;">
+</div>
+<div id="target4" style="left:450px; background:orange;position: absolute; width:100px; height:100px;">
+</div>
+<input id="input" style="left:450px; top: 200px; position: absolute;">
+<script>
+    window.eventStorage = [];
+
+    // Overlapping target test
+
+    var target1 = document.querySelector('#target1');
+    var element = document.querySelector('#element');
+
+    target1.addEventListener('mousedown', function () {
+        element.style.top = '0px';
+
+        window.eventStorage.push('target mousedown');
+    });
+
+    target1.addEventListener('mouseup', function () {
+        window.eventStorage.push('target mouseup');
+    });
+
+    target1.addEventListener('click', function () {
+        window.eventStorage.push('target click');
```

Comment:
I suppose, it's better to specify what the target was clicked (change `target` to `target1`). 
Also you can unify this handlers:
function saveEventToStorage (e) {
    window.eventStorage.push(e.target.id + ' ' + e.type);
}

target1.addEventListener('mouseup', saveEventToStorage);
element.addEventListener('mousedown', saveEventToStorage); etc...